src/moodle.rs

```diff
@@ -82,3 +85,144 @@ pub fn create_exam(
+        let prefix5 = "task5_prefix".to_string();
+        let prefix6 = "task6_prefix".to_string();
+
+        let flag1 = Flag::new_random_flag(taskid, 32);
+        let flag2 = Flag::new_random_flag(taskid2, 32);
+        let flag3 = Flag::new_random_flag(taskid3, 32);
+        let flag4 = Flag::new_user_flag(prefix4, &Algorithm::HMAC_SHA3_256, &secret, &taskid4, &id);
+        let flag5 =
+            Flag::new_user_flag(prefix5, &Algorithm::HMAC_SHA3_256, &secret2, &taskid5, &id);
+        let flag6 =
+            Flag::new_user_flag(prefix6, &Algorithm::HMAC_SHA3_256, &secret3, &taskid6, &id);
+
+        flags2.push(flag1);
+
+        flags3.push(flag2);
+        flags3.push(flag3);
+
+        flags.push(flag4);
+        flags.push(flag5);
+        flags.push(flag6);
+
+        let answers = process_multiple_flags(flags, " ");
+        let answers2 = process_multiple_flags(flags2, " ");
+        let answers3 = process_multiple_flags(flags3, " ");
+
+        for answer in answers {
+            match answer.fraction {
+                33 => {
+                    assert!(
+                        answer.text.contains("task4_prefix:")
+                            || answer.text.contains("task5_prefix:")
+                            || answer.text.contains("task6_prefix:")
+                    );
+                }
+                67 => {
+                    assert!(
+                        (answer.text.contains("task4_prefix:")
+                            && answer.text.contains("task5_prefix:"))
+                            || (answer.text.contains("task6_prefix:")
+                                && answer.text.contains("task5_prefix:"))
+                            || (answer.text.contains("task6_prefix:")
+                                && answer.text.contains("task4_prefix:"))
+                    );
+                }
+                100 => {
+                    assert!(
+                        (answer.text.contains("task4_prefix:")
+                            && answer.text.contains("task5_prefix:")
+                            && answer.text.contains("task6_prefix:"))
+                    );
+                }
+                _ => {
+                    unreachable!("Unexpected fraction value encountered in test")
+                }
+            }
+        }
+        for answer in answers2 {
+            assert!(answer.fraction == 100);
+            assert!(answer.text.contains("task1:"));
+        }
+        for answer in answers3 {
+            match answer.fraction {
+                50 => {
+                    assert!(answer.text.contains("task2:") || answer.text.contains("task3:"));
+                }
+                100 => {
+                    assert!(answer.text.contains("task2:") && answer.text.contains("task3:"));
+                }
+                _ => {
+                    unreachable!("Unexpected fraction value encountered in test")
+                }
+            }
+        }
+    }
+}
```

Comment:
OK. This does not ensure that there are no duplicates but maybe we can trust the logic of itertools. However, tests should always ensure that if we made changes in the code, the end functionality remains and we get confidence that system still works, so tests should be strict enough.